apps/server/src/shared/domain/entity/user.entity.ts

```diff
@@ -156,11 +156,38 @@ export class User extends BaseEntityWithTimestamps implements EntityWithSchool {
 			permissions = [...permissions, ...rolePermissions];
 		});
 
-		const uniquePermissions = [...new Set(permissions)];
+		const setOfPermission = this.checkSchoolPermissions(permissions, roles);
+
+		const uniquePermissions = [...setOfPermission];
 
 		return uniquePermissions;
 	}
 
+	private checkSchoolPermissions(permissions: string[], roles: Role[]) {
```

Comment:
we have some duplicate code with service from me module....is not easy to reuse it here, but a TODO comment, both here and in the me service, would make it easier to find it if we make changes in the future.